backend/src/main/java/com/handong/rebon/shop/application/CategoryService.java

```diff
@@ -0,0 +1,56 @@
+package com.handong.rebon.shop.application;
+
+import com.handong.rebon.exception.category.CategoryExistException;
+import com.handong.rebon.exception.category.CategoryNoParentException;
+import com.handong.rebon.shop.application.dto.CategoryRequestDto;
+import com.handong.rebon.shop.domain.category.Categories;
+import com.handong.rebon.shop.domain.category.Category;
+import com.handong.rebon.shop.domain.repository.CategoryRepository;
+
+import lombok.RequiredArgsConstructor;
+
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Service
+public class CategoryService {
+
+    private final CategoryRepository categoryRepository;
+
+    public Long create(String name) {
+        isCategoryExist(name);
+
+        Category newCategory = Category.builder()
+                                       .name(name)
+                                       .children(new Categories())
+                                       .build();
+        categoryRepository.save(newCategory);
+        return newCategory.getId();
+    }
+
+    public Long create(CategoryRequestDto categoryRequestDto) {
+
+        Category parent = categoryRepository.findById(categoryRequestDto.getParentId())
+                                            .orElseThrow(CategoryNoParentException::new);
+
+        String categoryName = categoryRequestDto.getName();
+
+        Category newCategory = Category.builder()
+                                       .name(categoryName)
+                                       .children(new Categories())
+                                       .build();
+
+        parent.addChildCategory(newCategory);
+        categoryRepository.save(newCategory);
+
+        return newCategory.getId();
+    }
+
+    private void isCategoryExist(String name) {
+        if(categoryRepository.existsByName(name)) throw new CategoryExistException();
```

Comment:
if문 한 줄이어도 {}를 붙여주는 게 일관성있고 가독성 면에서도 더 좋을 것 같아요
```suggestion
        if(categoryRepository.existsByName(name)) {
            throw new CategoryExistException();
        }
```